helios-integrations/helios-retrofit/src/main/kotlin/helios/retrofit/HeliosResponseBodyConverter.kt

```diff
@@ -13,8 +15,7 @@ class HeliosResponseBodyConverter<T>(private val decoder: Decoder<T>) : Converte
     Json
       .parseFromByteBuffer(ByteBuffer.wrap(value.byteStream().readBytes()))
       .flatMap { it.decode(decoder) }
-      .getOrElse { null }
-      .also { value.close() }
+      .flatMap { Try { value.close() }.toEither().`as`(it) }.getOrElse { null }
```

Comment:
This isn't a very cash money implementation of safe resource acquisition, and there's the top-level try/catch affecting it too.

Reply:
Soo, is it better to leave it as it was or should I remove the try/catch and keep it this way? If I'm honest I prefer this block as it was before. It's easier to read than another `flatMap` 🤔

Reply:
Oh sorry I missed the resource acquisition, yes it would be better to leave at it was but left a `TODO` to replace it when `Arrow` support the `Resource` type class